src/qililab/calibration/calibration_node.py

```diff
@@ -0,0 +1,668 @@
+                ...
+
+                # Set the platform:
+                platform = ql.build_platform(path=platform_path)
+                platform.connect()
+                platform.initial_setup()
+                platform.turn_on_instruments()
+
+                # Define circuit
+                circuit = ...
+
+                # Loop over the sweeps executing the platform:
+                results_list = []
+                for X in sweep_interval:
+                    platform.set_parameter(alias=alias, parameter=ql.Parameter.X, value=X)
+                    result = platform.execute(program=circuit, num_avg=hw_avg, repetition_duration=repetition_duration)
+                    results_list.append(result.array)
+
+                results = np.hstack(results_list)
+
+            |
+
+            **3) An analysis procedure**, that plots and fits the obtained data to the expected theoretical behaviour and finds the optimal desired parameters.
+
+            .. code-block:: python
+
+                def fit(xdata, results):
+                    ...
+
+                fitted_values, x_data, y_data, figure = fit(xdata=sweep_interval, results=results)
+                plt.show()
+
+            |
+
+            **4) An export data cell**, that calls ``export_calibration_outputs()`` with the dictionary to retrieve from the notebook into the calibration workflow:
+
+            .. code-block:: python
+
+                from qililab.automatic_calibration.calibration_node import export_calibration_outputs
+
+                export_calibration_outputs(
+                    {
+                        "check_parameters": {"x": sweep_interval, "y": results},
+                        "platform_params": [(bus_alias0, qubit, param_name0, fitted_values[0]), (bus_alias1, qubit, param_name1, fitted_values[1])],
+                        "fidelities": [(qubit, "fidelity1", 0.9), (qubit, "fidelity2", 0.95)]  # Fidelities in the output dictionary are optional.
+                    }
+                )
+
+            where the ``check_parameters`` are a dictionary of the saved results to do comparisons against. And the ``platform_params`` are a list of parameters to set on the platform.
+    """
+
+    def __init__(
+        self,
+        nb_path: str,
+        in_spec_threshold: float,
+        bad_data_threshold: float,
+        comparison_model: Callable,
+        drift_timeout: float,
+        qubit_index: int | list[int] | None = None,
+        node_distinguisher: int | str | None = None,
+        input_parameters: dict | None = None,
+        sweep_interval: np.ndarray | None = None,
+        number_of_random_datapoints: int = 10,
+    ):
+        if in_spec_threshold > bad_data_threshold:
+            raise ValueError("`in_spec_threshold` must be smaller or equal than `bad_data_threshold`.")
+
+        if len(nb_path.split("\\")) > 1:
+            raise ValueError("`nb_path` must be written in unix format: `folder/subfolder/.../file.ipynb`.")
```

Comment:
We can use the `pathlib` library, which contains a `Path` class that is agnostic of the path format

Reply:
I'll take a look 🙌

Reply:
Moving this to a new task, to merge the PR 🚀